Library/Homebrew/extend/os/mac/requirements/java_requirement.rb

```diff
@@ -1,13 +1,28 @@
 class JavaRequirement < Requirement
-  cask "java"
+  def initialize(tags)
+    @version = tags.shift if /(\d+\.)+\d/ =~ tags.first
+    @cask = cask
+    super
+  end
 
   env do
     env_java_common
     env_oracle_jdk || env_apple
   end
 
+  def cask
+    which_cask_needed?
```

Comment:
Any reason `which_cask_needed?` can't fill this method?

Reply:
Nope, good idea! 👍